heat/core/tiling.py

```diff
@@ -0,0 +1,756 @@
+                torch.tensor(row_inds, device=arr._DNDarray__array.device)
+            ).flatten():
+                row_inds_hold.append(row_inds[i.item()])
+            row_inds = row_inds_hold
+
+        tile_map = torch.zeros(
+            [len(row_inds), len(col_inds), 3], dtype=torch.int, device=arr._DNDarray__array.device
+        )
+        # if arr.split == 0:  # adjust the 1st dim to be the cumsum
+        col_inds = [0] + col_inds[:-1]
+        col_inds = torch.tensor(col_inds, device=arr._DNDarray__array.device).cumsum(dim=0)
+        # if arr.split == 1:  # adjust the 0th dim to be the cumsum
+        row_inds = [0] + row_inds[:-1]
+        row_inds = torch.tensor(row_inds, device=arr._DNDarray__array.device).cumsum(dim=0)
+
+        for num, c in enumerate(col_inds):  # set columns
+            tile_map[:, num, 1] = c
+        for num, r in enumerate(row_inds):  # set rows
+            tile_map[num, :, 0] = r
+
+        for i in range(arr.comm.size):
+            st = sum(row_per_proc_list[:i])
+            sp = st + row_per_proc_list[i]
+            tile_map[..., 2][st:sp] = i
+        # to adjust if the last process has more tiles
+        i = arr.comm.size - 1
+        tile_map[..., 2][sum(row_per_proc_list[:i]) :] = i
+
+        if arr.split == 1:
+            st = 0
+            for pr, cols in enumerate(col_per_proc_list):
+                tile_map[:, st : st + cols, 2] = pr
+                st += cols
+
+        for c, i in enumerate(row_per_proc_list):
+            try:
+                row_per_proc_list[c] = i.item()
+            except AttributeError:
+                pass
+        for c, i in enumerate(col_per_proc_list):
+            try:
+                col_per_proc_list[c] = i.item()
+            except AttributeError:
+                pass
+
+        # =========================================================================================
+        self.__DNDarray = arr
+        self.__col_per_proc_list = (
+            col_per_proc_list if arr.split == 1 else [len(col_inds)] * len(col_per_proc_list)
+        )
+        self.__lshape_map = lshape_map
+        self.__last_diag_pr = last_diag_pr.item()
+        self.__row_per_proc_list = (
+            row_per_proc_list if arr.split == 0 else [len(row_inds)] * len(row_per_proc_list)
+        )
+        self.__tile_map = tile_map
+        self.__row_inds = list(row_inds)
+        self.__col_inds = list(col_inds)
+        # =========================================================================================
+
+    @property
+    def arr(self):
+        """
+        Returns
+        -------
+        DNDarray : the DNDarray for which the tiles are defined on
+        """
+        return self.__DNDarray
+
+    @property
+    def col_indices(self):
+        """
+        Returns
+        -------
+        list : list containing the indices of the tile columns
+        """
+        return self.__col_inds
+
+    @property
+    def lshape_map(self):
```

Comment:
Do we want to over either the lmap and/or the tile map via the iterator interface, aka `__iter__()`?

Reply:
both of these are torch tensors, so shouldnt the iterator be simply the torch iterator? 

in what scenarios do you anticipate the need for the iterator interface?